src/web/middleware.rs

```diff
@@ -274,6 +288,93 @@ where
     }
 }
 
+pub struct SentryWrapper;
+
+impl SentryWrapper {
+    pub fn new() -> Self {
+        SentryWrapper::default()
+    }
+}
+
+impl Default for SentryWrapper {
+    fn default() -> Self {
+        Self
+    }
+}
+
+impl<S, B> Transform<S> for SentryWrapper
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error> + 'static,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type InitError = ();
+    type Transform = SentryWrapperMiddleware<S>;
+    type Future = FutureResult<Self::Transform, Self::InitError>;
+
+    fn new_transform(&self, service: S) -> Self::Future {
+        future::ok(SentryWrapperMiddleware {
+            service: Rc::new(RefCell::new(service)),
+        })
+    }
+}
+
+#[derive(Debug)]
+pub struct SentryWrapperMiddleware<S> {
+    service: Rc<RefCell<S>>,
+}
+
+impl<S, B> Service for SentryWrapperMiddleware<S>
+where
+    S: Service<Request = ServiceRequest, Response = ServiceResponse<B>, Error = Error> + 'static,
+    S::Future: 'static,
+    B: 'static,
+{
+    type Request = ServiceRequest;
+    type Response = ServiceResponse<B>;
+    type Error = Error;
+    type Future = Box<dyn Future<Item = Self::Response, Error = Self::Error>>;
+
+    fn poll_ready(&mut self) -> Poll<(), Self::Error> {
+        self.service.poll_ready()
+    }
+
+    fn call(&mut self, sreq: ServiceRequest) -> Self::Future {
+        let tags = Tags::from_request_head(sreq.head());
+        sreq.extensions_mut().insert(tags.clone());
+
+        Box::new(self.service.call(sreq).and_then(move |resp| {
+            // handed an actix_error::error::Error;
+            // Fetch out the tags (in case any have been added.)
+            // Tags associated with the sreq.extensions() may not
+            // always be preserved. Use the orignal set of
+            // tags and overwrite with any additional tags that may
+            // have been set in the response.
+            let mut tag_vals = tags.unwrap_or_default();
+            if let Some(t) = resp.response().extensions().get::<Tags>() {
+                for (k, v) in t.tags.clone() {
+                    tag_vals.tags.insert(k, v);
+                }
+            };
+            match resp.response().error() {
```

Comment:
We currently only capture errors if they're 500 responses, we should continue restricting what we capture to only things we care about (is there anything else other than 500s? probably not).

Also this PR should remove the old capturing in db/error.rs so it's only done once.